enterprise_access/apps/subsidy_access_policy/content_metadata_api.py

```diff
@@ -79,3 +81,46 @@ def get_and_cache_catalog_contains_content(enterprise_catalog_uuid, content_key,
     )
     TieredCache.set_all_tiers(cache_key, result, timeout or DEFAULT_CACHE_TIMEOUT)
     return result
+
+
+def get_list_price_for_content(enterprise_customer_uuid, content_key, content_metadata=None):
+    """
+    Given a customer and content identifier, fetch content metadata and return a list price
+    dictionary. If the caller already has a dictionary of ``content_metadata`` in scope, this
+    function computes its return value from that.
+    Returns:
+        A dictionary of the form
+        ```
+        {
+            "usd": 149.50, # the list price in US Dollars as a float
+            "usd_cents": 14950 # the list price in USD Cents as an int
+        }
+
+    Raises:
+        A ``ContentPriceNullException`` if we encountered an HTTPError fetch content metadata.
+    """
+    if not content_metadata:
+        try:
+            content_metadata = get_and_cache_content_metadata(enterprise_customer_uuid, content_key)
+        except requests.exceptions.HTTPError as exc:
+            logger.warning(
+                f'{exc} when checking content metadata for {enterprise_customer_uuid} and {content_key}'
+            )
+            raise ContentPriceNullException(f'Could not determine list price for {content_key}') from exc
+
+    # Note that the "content_price" key is guaranteed to exist, but the value may be None.
+    return list_price_dict_from_usd_cents(content_metadata['content_price'])
+
+
+def list_price_dict_from_usd_cents(list_price_integer_cents):
+    """
+    Helper to compute a list price dictionary given the non-negative price of the content in USD cents.
+    """
+    list_price_decimal_dollars = None
+    if list_price_integer_cents is not None:
+        list_price_decimal_dollars = float(list_price_integer_cents) / 100
```

Comment:
The float cast here concerns me since we are working with currency. Why not `Decimal`?

Reply:
This is how I found it, Chief! ;)
We could work with a decimal, here, but it'll eventually get `float`-ed when it's serialized as JSON for view resonses.  And the list price is generally used for display purposes only.  I'll try it out and see how it looks.

Reply:
hmm, ok `s/float/Decimal/` seems to "just work", so we'll go with that.